### PR TITLE
[GSoC] Coroutines migration ConfSetSubdecks

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Fixes
Fixes a part of #7108

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 1]
```diff
@@ -295,7 +296,22 @@ class DeckOptions :
                                 }
                             }
                             "confSetSubdecks" -> if (value as Boolean) {
-                                TaskManager.launchCollectionTask(CollectionTask.ConfSetSubdecks(deck, mOptions), confChangeHandler())
+                                launch(getCoroutineExceptionHandler(this@DeckOptions)) {
+                                    preConfChange()
+                                    try {
```
Comment (ShridharGoel) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:301]: Is there any use of `getCoroutineExceptionHandler` if this is already wrapped inside `try`?

**Review by ShridharGoel — APPROVED** [review]

**Comment by divyansh-dxn** [comment]
conflicts...on it.

**Comment by lukstbit** [comment]
@divyansh-dxn Still has conflicts...

**Review by Arthur-Milchior — APPROVED** [review]

**Comment by Arthur-Milchior** [comment]
Rebased

**Comment by divyansh-dxn** [comment]
Thanks, @Arthur-Milchior for fixing the conflicts!

**Review by lukstbit — APPROVED** [review]

**Pull request merged**